sample-code/src/orchestration.ts

```diff
@@ -482,3 +484,60 @@ export async function orchestrationChatCompletionImage(): Promise<OrchestrationR
     }
   });
 }
+
+/**
+ * Response type that is guaranteed to be respected by the orchestration LLM response.
+ */
+export interface TranslationResponseType {
+  /**
+   * The language of the translation, randomly chosen by the LLM.
+   */
+  language: string;
+  /**
+   * The translation of the input sentence.
+   */
+  translation: string;
+}
+/**
+ * Ask about the image content using a template.
+ * @returns The orchestration service response.
+ */
+export async function orchestrationResponseFormat(): Promise<TranslationResponseType> {
+  const translationSchema = z
+    .object({
+      language: z.string(),
+      translation: z.string()
+    })
+    .strict();
+  const orchestrationClient = new OrchestrationClient({
+    llm,
+    templating: {
+      template: [
+        {
+          role: 'system',
+          content:
+            'You are a helpful AI that translates simple sentences into different languages. The user will provide the sentence. You then choose a language at random and provide the translation.'
+        },
+        {
+          role: 'user',
+          content: '{{?input}}'
+        }
+      ],
+      response_format: {
+        type: 'json_schema',
+        json_schema: {
+          name: 'translation_response',
+          strict: true,
+          schema: zodToJsonSchema(translationSchema)
+        }
+      }
+    }
+  });
+
+  const resonse = await orchestrationClient.chatCompletion({
+    inputParams: {
+      input: 'Hello World! Why is this phrase so famous?'
+    }
+  });
+  return JSON.parse(resonse.getContent()!) as TranslationResponseType;
```

Comment:
[req] If I remember this correctly, a zod schema can be used to parse (aka validate) any object. `translationSchema.parse(response.getContent())` something like this. Can you have a try? Or have a look at `tests/schema-tests/test`

Reply:
translationSchema.parse(response.getContent()) doesnt work since parse expects an object which means i have to do translationSchema.parse(JSON.parse(resonse.getContent()!))